internal/handlers/user.go

```diff
@@ -0,0 +1,429 @@
+		http.Error(w, "Failed to register user", http.StatusInternalServerError)
+		_ = tx.Rollback(uh.OrderService.OrderRepository.DBStorage.Ctx)
+		return
+	}
+
+	if err = uh.UserBalanceService.CreateUserBalance(user); err != nil {
+		http.Error(w, "Failed to register user", http.StatusInternalServerError)
+		_ = tx.Rollback(uh.OrderService.OrderRepository.DBStorage.Ctx)
+		return
+	}
+
+	if err := tx.Commit(uh.OrderService.OrderRepository.DBStorage.Ctx); err != nil {
+		http.Error(w, "Failed to commit transaction", http.StatusInternalServerError)
+		return
+	}
+
+	token, err := generateToken(user)
+
+	if err != nil {
+		http.Error(w, "Failed to create token", http.StatusInternalServerError)
+		return
+	}
+
+	http.SetCookie(w, &http.Cookie{
+		Name:     "token",
+		Value:    token,
+		HttpOnly: true,
+		Path:     "/",
+		MaxAge:   3600,
+	})
+
+	w.Header().Set("Authorization", token)
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(map[string]string{"message": "User registered successfully"})
+}
+
+func (uh *UserHandler) Login(w http.ResponseWriter, r *http.Request) {
+	var creds middleware.Credentials
+
+	err := json.NewDecoder(r.Body).Decode(&creds)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	authUser, err := uh.UserService.AuthenticateUser(creds.Username, creds.Password)
+
+	if err != nil {
+		http.Error(w, "Unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	token, err := generateToken(authUser)
+	if err != nil {
+		http.Error(w, "Failed to create token", http.StatusInternalServerError)
+		return
+	}
+
+	http.SetCookie(w, &http.Cookie{
+		Name:     "token",
+		Value:    token,
+		HttpOnly: true,
+		Path:     "/",
+		MaxAge:   3600,
+	})
+
+	w.Header().Set("Authorization", token)
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(map[string]string{"message": "Login successful"})
+}
+
+func generateToken(user models.User) (string, error) {
+	claims := jwt.MapClaims{
+		"username": user.Username,
+		"exp":      time.Now().Add(time.Hour * 72).Unix(),
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	return token.SignedString([]byte(middleware.SecretKey))
+}
+
+func (uh *UserHandler) SaveOrder(w http.ResponseWriter, r *http.Request) {
+	username := r.Context().Value(middleware.UsernameKey).(string)
+
+	userID := uh.UserService.UserRepository.IsUserExists(username)
+
+	if userID < 0 {
+		http.Error(w, "пользователь не найден", http.StatusNotFound)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+
+	if err != nil {
+		http.Error(w, "Could not read body", http.StatusBadRequest)
+		return
+	}
+
+	defer r.Body.Close()
+
+	orderNumber := string(body)
+	isDigit := isDigits(orderNumber)
+
+	if !ValidateNumber(orderNumber) {
+		http.Error(w, "Неверный формат номера заказа", http.StatusUnprocessableEntity)
+		return
+	}
+
+	if !isDigit {
+		http.Error(w, "Неверный формат запроса", http.StatusBadRequest)
+		return
+	}
```

Comment:
Мне кажется стоит упорядочить вызовы:
```suggestion
	orderNumber := string(body)
	isDigit := isDigits(orderNumber)
	if !isDigit {
		http.Error(w, "Неверный формат запроса", http.StatusBadRequest)
		return
	}

	if !ValidateNumber(orderNumber) {
		http.Error(w, "Неверный формат номера заказа", http.StatusUnprocessableEntity)
		return
	}

```